zerver/views/registration.py

```diff
@@ -881,6 +886,75 @@ def redirect_to_email_login_url(email: str) -> HttpResponseRedirect:
     return HttpResponseRedirect(redirect_url)
 
 
+@typed_endpoint
+def realm_import_status(
+    request: HttpRequest,
+    maybe_user_profile: UserProfile | AnonymousUser,
+    *,
+    confirmation_key: str,
+) -> HttpResponse:
+    try:
+        # We don't use get_object_from_key because we're fine if it's been marked used
+        confirmation = Confirmation.objects.get(
+            confirmation_key=confirmation_key, type__in=[Confirmation.REALM_CREATION]
+        )
+    except Confirmation.DoesNotExist:
+        raise JsonableError(_("Unauthenticated"))
+
+    preregistration_realm = confirmation.content_object
+    assert isinstance(preregistration_realm, PreregistrationRealm)
+    try:
+        realm = Realm.objects.get(string_id=preregistration_realm.string_id)
+    except Realm.DoesNotExist:
+        # XXX Seplunk the filesystem for the converted import to snoop
+        # on which files have been written?  Only one dir should match
+        # the tempdir glob
```

Comment:
This comment should probably be updated to say:

> TODO: Either store the path to the temporary conversion directory on `preregistration_realm.data_import_metadata`, or have the conversion process support writing updates to this for a better progress indicator.

Reply:
Updated! Thanks!